modules/client/src/main/java/org/apache/ignite/internal/client/compute/ClientCompute.java

```diff
@@ -271,16 +274,12 @@ private CompletableFuture<PayloadInputChannel> executeOnOneNode(
         return ch.serviceAsync(
                 ClientOp.COMPUTE_EXECUTE,
                 w -> {
-                    if (w.clientChannel().protocolContext().clusterNode().name().equals(node.name())) {
-                        w.out().packNil();
-                    } else {
-                        w.out().packString(node.name());
-                    }
-
+                    String[] nodeNames = nodes.stream().map(ClusterNode::name).toArray(String[]::new);
+                    w.out().packObjectArrayAsBinaryTuple(nodeNames);
```

Comment:
1. Java streams are unnecessary here (and inefficient)
2. `packObjectArrayAsBinaryTuple` is excessive here, it includes type code for every element

Instead, let's do `packInt` for size and then `packString` in a loop.